### DISCUSSION TITLE
midwayjs 框架 推入bull 后 ctx 上下文 没有 safeCurl 和 curl 方法 （不使用 axios等包）能否直接使用内置方法

### DISCUSSION THREAD

**Discussion by mynamexjl** [midwayjs/midway#3356] [Q&A] [answered]
midwayjs 框架 推入bull 后 ctx 上下文 没有 safeCurl 和 curl 方法 （不使用 axios等包）能否直接使用内置方法

**Comment by czy88840616** [comment]
curl 那是egg的逻辑，你需要 curl 可以从 egg 的 app 上拿
```
@App('egg')
app;
```

midway 有自己的 [makeHttpRequest](https://midwayjs.org/docs/extensions/axios) 方法。

**Comment by mynamexjl** [comment]
业务需求必须使用 safecurl 请求   
如果用 midway 自有的 makeHttpRequest  是否也有 类似 security 规则可以使用
在bull 队列中依赖注入后egg 直接队列消费就阻塞了
@App('egg')
app

通过 在add的时候把 ctx 透传给队列  直接将方法转成了对象  方法体都不能使用

**Comment by czy88840616** [comment]
用egg的app就有egg自己的规则了吧？

**Comment by mynamexjl** [comment]
![image](https://github.com/midwayjs/midway/assets/25628679/cd8a99ef-b9ca-42c8-a50b-f9d768803bc5)

**Comment by mynamexjl** [comment]
现在就是。在bull 队列中注入egg后 使用内部方法。无法执行  没有任何输出。 队列也不会去消费

**Comment by mynamexjl** [comment]
把 egg 去掉后。队列又能正常消费

**Comment by czy88840616** [comment, marked as answer]
![image](https://github.com/midwayjs/midway/assets/418820/e6a855f9-e42e-4fcb-ac57-090a68f54e16)

**Comment by mynamexjl** [comment]
🙏🙏🙏        
问题解决了
